psi4/driver/driver_cbs.py

```diff
@@ -917,135 +1137,468 @@ def return_energy_components():
+        for just the correlation component, None if no extrapolation requested.
+    bdata
+        Names and zetas of the basis sets to extrapolate from.
+    funcname
+        Functional name for DFT extrapolation.
+
+    Returns
+    -------
+    alpha : Union[float, None]
+        Extrapolation alpha to be used.
+    """
+    nbasis = len(bdata[0])
+    if nbasis == 1:
+        return None
+    elif nbasis == 2:
+        if xtpl_type == "scf":
+            return 1.63  # this is the scf_xtpl_helgaker_2 default
+        elif xtpl_type == "corl":
+            return 3.00  # this is the corl_xtpl_helgaker_2 default
+        elif xtpl_type in ["fctl", "dh", "dft"]:
+            return _get_dfa_alpha(xtpl_type, bdata, funcname)
+        else:
+            raise ValidationError(
+                f"Stage treatment must be one of ['scf','corl','fctl','dh',"
+                f"'dft'], not '{xtpl_type}'"
+            )
+    elif nbasis == 3:
+        return None  # we don't need alpha for _3 methods
+
+
+def _interpret_cbs_inputs(
+    cbs_metadata: List[dict], 
+    molecule: Union[qcdb.molecule.Molecule, core.Molecule],
+) -> List[dict]:
     """ A helper function which validates the ``cbs_metadata`` format,
     expands basis sets, and provides sensible defaults for optional arguments.
 
+    There are currently three ways to run a CBS calculation:
+
+    1) By specifying arguments, such as ``scf_wfn``, ``corl_basis``, or ``delta_scheme``.
+       This is the original interface, however option (3) is preferred nowadays.
+    2) By using the "method/basis+delta" syntax. This option offeres limited features,
+       as the default treatments and alphas are imposed.
+    3) By explicitly passing ``cbs_metadata`` as one of the arguments. This is the
+       full-featured user-facing interface.
+
+    These three are interpreted into a single, internal specification here. For the
+    description of the user-facing interfaces see :py:func:`~psi4.cbs`.
+
+    The main role of this function is to interpret the user-provided arguments,
+    infer any missing stages (if possible) such as the SCF stage if the user
+    provides "MP2/cc-pv[dt]z" or the split stages if user requests an extrapolation
+    of a DFA.
+
+    The resulting ``metadata`` :class:`(list[dict])` contains the list of all stages,
+    with the following entries set in each of the dicts:
+
+     - ``"wfn"`` :class:`(str)`: method name for upper level of theory (l.o.t.)
+     - ``"basis"`` :class:`(tuple)`: definition of the basis sets for the upper
+       l.o.t., provided as an "expanded" tuple of names and zeta levels
+     - ``"options"`` :class:`(dict)`: specification of additional Psi4 options to
+       switch on in the upper l.o.t.
+     - ``"component"`` :class:`(str)`: component of the total E/G/H to use for
+       extrapolation as the upper l.o.t.; for WFT it's usually the same as ``"wfn"``,
+       for DFT it allows splitting of the total DFT energy into its components. 
```

Comment:
Is there anything DFT-specific here? In principle, you could do the same with a "wavefunction" method, right? The example I have in mind is getting a (T) delta from CCSD(T) and adding that to a LCCSD calculation. Is this part of the interface how you would do that?